kube-derive/src/lib.rs

```diff
@@ -327,6 +327,41 @@ pub fn derive_custom_resource(input: proc_macro::TokenStream) -> proc_macro::Tok
     custom_resource::derive(proc_macro2::TokenStream::from(input)).into()
 }
 
+/// Generates a JsonSchema patch with a set of CEL expression validation rules applied on the CRD.
+///
+/// # Example
+///
+/// ```rust
+/// use kube::Validated;
+/// use kube::CustomResource;
+/// use serde::Deserialize;
+/// use serde::Serialize;
+/// use schemars::JsonSchema;
+/// use kube::core::crd::CustomResourceExt;
+///
+/// #[derive(CustomResource, Validated, Serialize, Deserialize, Clone, Debug, JsonSchema)]
+/// #[kube(group = "kube.rs", version = "v1", kind = "Struct")]
+/// struct MyStruct {
+///     #[serde(default = "default")]
+///     #[validated(rule = Rule{rule: "self != ''".into(), message: Some("failure message".into()), ..Default::default()})]
```

Comment:
pretty cool that you can do a full struct here like this. does it support shorthands as well if we were to create builders on `Rule`? could lead to:
```rust
#[validated(rule = Rule::rule("self != ''").message("failure message")]
```
possibly

Reply:
it is possible, including something simple such as `#[validated(rule = “self != ‘’”.into())]`.